ports lethalstation's [genemodded] quirk! allows roundstart selection of ONE (only ONE!) genetics mutation

CandleJaxx opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

okay so this basically does what it says on the tin. a positive quirk that lets you select ONE genetics mutation (if your species is eligible, sorry robots / plasmamen) to have by default

## Why It's Good For The Game

![image](https://github.com/user-attachments/assets/5539d52e-2425-4056-9e42-f5f5f86aa105) gigantism example for those ppl who want to be EVEN BIGGER at the cost of moving slower than a snailperson

other selectable quirks include, (but are not limited to,):

pressure adaptation! 

epilepsy!

chameleon!

thermal vision!

h.a.r.s.!

and last but not least,

chav!

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and its effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: added genemodded quirk
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

CandleJaxx commented:
so heads up for you more codertypes, there are some genes here that might be good to be edited out- but i think this should be ready for a testmerge

kittysmooch commented:
free loadout thermals is pretty insane even for a noted anti-balancejak like me i think

CandleJaxx commented:
> free loadout thermals is pretty insane even for a noted anti-balancejak like me i think

yeah i'd open u to make a list of specific ones 2 remove but also: i dont know how to do that beyond using the inbuilt var/locked to just prevent them from showing up

bearrrrrrrr commented:
some of these just are not balanced for roundstart. loadout thermals and chameleon is crazy. void magnet is also, like, near-antag stuff that lets u become literally invincible for a second or two and is super disruptive for rp (since iirc it literally just flings you around sometimes)

suggest maybe just making it so that you can only pick the Minor tagged stuff? ppl say 'balancejak' but i don't think basically removing genetics is the intention here

CandleJaxx commented:
> some of these just are not balanced for roundstart. loadout thermals and chameleon is crazy. void magnet is also, like, near-antag stuff that lets u become literally invincible for a second or two (gives u lavaland disappear talisman) and is super disruptive for rp (since iirc it literally just flings you around sometimes)
> 
> suggest maybe just making it so that you can only pick the Minor tagged stuff? ppl say 'balancejak' but i don't think basically removing genetics is the intention here

agree! some def need to be edited. i do not know how to do that at the moment however

bearrrrrrrr commented:
ahhhh, a code issue, then? easy. try uhh _mumble mumble_ global var list _mumble mumble_ .tsx _mumble_

<- (can't code)

CandleJaxx commented:
blacklist added, removed thermal vision, amongst some other useless quirks that didn't have an actual impact. removed telepathy and changed it to a bespoke quirk, but this may be reverted with the upcoming grey species- i encourage staff / players to use quirks like telepathy sparingly. this playerbase SHOULD be able to be trusted with options while not minmaxxing. if players choose to abuse this (taking telepathy for no real reason, trying to powercreep, etc) it'll probably be removed / reverted

CandleJaxx commented:
oh if maints want any other mutations added to blacklist, lemme know

CandleJaxx commented:
making radiation mutation visible actually bc androids dont glow when rad'd and i dont want to stealth nuke people- i WILL work in a toggle on this eventually
